language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureInfoModelBuilder.java

```diff
@@ -0,0 +1,342 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.signature;
+
+import io.ballerina.compiler.api.symbols.Documentable;
+import io.ballerina.compiler.api.symbols.Documentation;
+import io.ballerina.compiler.api.symbols.FunctionSymbol;
+import io.ballerina.compiler.api.symbols.FunctionTypeSymbol;
+import io.ballerina.compiler.api.symbols.ParameterSymbol;
+import io.ballerina.compiler.api.symbols.RecordFieldSymbol;
+import io.ballerina.compiler.api.symbols.RecordTypeSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.VariableSymbol;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.SignatureContext;
+import org.ballerinalang.langserver.util.MarkupUtils;
+import org.eclipse.lsp4j.MarkupContent;
+import org.eclipse.lsp4j.ParameterInformation;
+import org.eclipse.lsp4j.SignatureInformation;
+import org.eclipse.lsp4j.SignatureInformationCapabilities;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.lsp4j.jsonrpc.messages.Tuple;
+import org.wso2.ballerinalang.compiler.util.Names;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A utility for building a SignatureInformation.
+ *
+ * @since 2201.0.3
+ */
+public class SignatureInfoModelBuilder {
+    private final Symbol symbol;
+    private final SignatureContext context;
+    private final List<ParameterInfoModel> parameterModels;
+
+    private ParameterInfoModel firstIncludedRecordParam;
+    private Either<String, MarkupContent> signatureDescription;
+    private final Map<String, String> paramsMap = new HashMap<>();
+
+    public SignatureInfoModelBuilder(Symbol symbol, SignatureContext context) {
+        this.symbol = symbol;
+        this.context = context;
+        this.parameterModels = new ArrayList<>();
+        fillParamInfoModels();
+    }
+
+    /**
+     * Build the possible list of signature information.
+     *
+     * @return {@link List} of SignatureInformation
+     */
+    public List<SignatureInformation> build() {
+        List<SignatureInformation> result = new ArrayList<>();
+
+        String functionName = getFunctionName();
```

Comment:
We can get rid of the variable and use the function call directly in the following line as it's the only usage